CMakeLists.txt

```diff
@@ -29,32 +28,90 @@ endmacro()
 option(BUILD_SHARED_LIBS "Build libraries as shared libraries" ON)
 message(STATUS "C++ Requests CMake Options")
 message(STATUS "=======================================================")
-cpr_option(USE_SYSTEM_CURL
-    "If ON, this project will look in the system paths for an installed curl library" OFF)
-cpr_option(BUILD_CPR_TESTS "Set to ON to build cpr tests." ON)
-cpr_option(BUILD_CPR_TESTS_SSL "Set to ON to build cpr ssl tests" ${BUILD_CPR_TESTS})
-cpr_option(GENERATE_COVERAGE "Set to ON to generate coverage reports." OFF)
+cpr_option(CPR_GENERATE_COVERAGE "Set to ON to generate coverage reports." OFF)
 cpr_option(CPR_CURL_NOSIGNAL "Set to ON to disable use of signals in libcurl." OFF)
-cpr_option(USE_SYSTEM_GTEST
-    "If ON, this project will look in the system paths for an installed gtest library" OFF)
-cpr_option(USE_OPENSSL "Use OpenSSL code. Experimental" ON)
-cpr_option(USE_WINSSL "Use WIN_SSL backend. Experimental" OFF)
+cpr_option(CPR_USE_SYSTEM_GTEST "If ON, this project will look in the system paths for an installed gtest library. If none is found it will use the build in one." OFF)
+cpr_option(CPR_FORCE_USE_SYSTEM_CURL "If enabled we will use the curl lib already installed on this system." OFF)
+cpr_option(CPR_ENABLE_SSL "Enables or disables the SSL backend. Required to perform HTTPS requests." ON)
+cpr_option(CPR_DETECT_SSL_BACKEND "Automatically detects the available SSL backends and uses the recommend for your system in case multiple are available. (WinSSL - Windows, OpenSSL - Linux, Mac, ...)" ON)
+cpr_option(CPR_FORCE_OPENSSL_BACKEND "Force to use the OpenSSL backend. For this to function set CPR_DETECT_SSL_BACKEND to OFF." OFF)
+cpr_option(CPR_FORCE_WINSSL_BACKEND "Force to use the WinSSL backend. For this to function set CPR_DETECT_SSL_BACKEND to OFF." OFF)
+cpr_option(CPR_BUILD_TESTS "Set to ON to build cpr tests." ON)
+cpr_option(CPR_BUILD_TESTS_SSL "Set to ON to build cpr ssl tests" ${CPR_BUILD_TESTS})
```

Comment:
I think it is a bad idea to rename the variables, since it will lead to problems in depending projects and especially package manager.

Reply:
Yes, but now we have a clear way to show which variables are offered by cpr.